Library/SharedFunctions.swift

```diff
@@ -106,7 +106,16 @@ internal func minAndMaxPledgeAmount(forProject project: Project, reward: Reward?
   case .none, .some(Reward.noReward):
     return (Double(country.minPledge ?? 1), Double(country.maxPledge ?? 10_000))
   case let .some(reward):
-    return (reward.minimum, Double(country.maxPledge ?? 10_000))
+    let min: Double
+
+    /// Account for the case where the originally selected reward pricing, for this backing, has since changed to late pledge pricing. We should always use the original pricing at the time of the backing over the most current state.
+    if project.personalization.backing?.isLatePledge == true {
+      min = reward.latePledgeAmount > 0 ? reward.latePledgeAmount : reward.minimum
+    } else {
+      min = reward.pledgeAmount > 0 ? reward.pledgeAmount : reward.minimum
+    }
+
+    return (min, Double(country.maxPledge ?? 10_000))
```

Comment:
I wanted to maintain the original `minimum` logic in the potential cases where the pledge amount or latePledgeAmounts haven't been set.

This could happen if a campaign is still live, doesn't have late pledge enabled, or rewards don't have a late pledge specific price set on them yet. I would hope that `pledgeAmount` always has a value, but I added a ternary for that case too just in case.

Reply:
That sounds good to me! Alternatively, if you want to ensure `pledgeAmount` (and even `latePledgeAmount`) always has values set, we could default them to `reward.minimum` if they're missing (instead of defaulting to 0). I don't have a strong opinion on this, though, so up to you.